tests/telemetry/test_telemetry.py

```diff
@@ -258,3 +281,240 @@ def test_virtualdb_table_streaming(duthosts, rand_one_dut_hostname, ptfhost, loc
+
+    Returns:
+        None.
+    """
+    status_result = duthost.critical_process_status(container_name)
+    if status_result["status"] is False or len(status_result["exited_critical_process"]) > 0:
+        return False
+
+    return True
+
+
+def postcheck_critical_processes(duthost, container_name):
+    """Checks whether the critical processes are running after container was restarted.
+
+    Args:
+        duthost: The AnsibleHost object of DuT.
+        container_name: Name of container.
+
+    Returns:
+        None.
+    """
+    logger.info("Checking the running status of critical processes in '{}' container ..."
+                .format(container_name))
+    is_succeeded = wait_until(CONTAINER_RESTART_THRESHOLD_SECS, CONTAINER_CHECK_INTERVAL_SECS, 0,
+                              check_critical_processes, duthost, container_name)
+    if not is_succeeded:
+        pytest.fail("Not all critical processes in '{}' container are running!"
+                    .format(container_name))
+    logger.info("All critical processes in '{}' container are running.".format(container_name))
+
+
+@pytest.fixture
+def test_mem_spike_setup_and_cleanup(duthosts, rand_one_dut_hostname, setup_streaming_telemetry, request):
+    """Customizes Monit configuration files before testing and restores them after testing.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: The fixture returns a randomly selected DuT.
+        setup_streaming_telemetry: Fixture to setup telemetry server authentication before testing.
+
+    Returns:
+        None.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    backup_monit_config_files(duthost)
+    customize_monit_config_files(duthost, request.param)
+    restart_monit_service(duthost)
+
+    yield
+
+    restore_monit_config_files(duthost)
+    restart_monit_service(duthost)
+
+
+@pytest.mark.disable_loganalyzer
+@pytest.mark.parametrize("test_mem_spike_setup_and_cleanup",
+                         ['    if status == 3 for 1 times within 2 cycles then exec "/usr/bin/restart_service telemetry" repeat every 2 cycles'],
+                         indirect=["test_mem_spike_setup_and_cleanup"])
+def test_mem_spike(duthosts, rand_one_dut_hostname, ptfhost, test_mem_spike_setup_and_cleanup):
+    """Test whether memory usage of telemetry container will increase and be restarted
+    or not by Monit if python gNMI client continuously creates channels with gNMI server.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: The fixture returns a randomly selected DuT.
+        pfthost: PTF docker binding to the selected DuT.
+        test_mem_spike_setup_and_cleanup: Fixture does testing setup and cleanup.
+
+    Returns:
+        None.
+    """
+    logger.info("Starting to test the memory spike issue of '{}' ...".format(CONTAINER_NAME))
+
+    duthost = duthosts[rand_one_dut_hostname]
+    dut_ip = duthost.mgmt_ip
+
+    skip_201911_and_older(duthost)
```

Comment:
@qiluo-msft I think your question `What prevent you to test 201911 image? Should we test 201911 image?` should be for the line 475, right?

I think this test case should run against 201911 image and will remove the line 475.